### ISSUE TITLE
DoctrineAnnotationBuilder doesnt generate form fields for identity columns / breaking database consistency

### ISSUE THREAD

**Issue by cbichis** [doctrine/DoctrineORMModule#372]
DoctrineAnnotationBuilder doesnt generate form fields for identity columns / breaking database consistency

Hello,

It seems currently the DoctrineAnnotationBuilder it doesnt generate form fields for identity columns even if @Annotation\Exclude is NOT set and there is defined a field type (hidden usually) for these columns.

This leads to issues, I can present an example for One-To-Many associations.

Lets suppose we have a Brand - Model as One-To-Many association. The entire code is created with annotations within entity classes, the Model been created as Composed object with is_collection and allow_add, allow_remove options.

We have populated the database like this:

Brand (id, name)
1 - Toyota

Model (id, name, brand)
1 - Yaris - 1
2 - Auris - 1
3 - Prius - 1
4 - Corolla - 1

Now, when we are removing the record #4 from database and add a new record with the name been "Other" the data assigned to the new record will be:

4 - Other - 1

This maybe is possibly to fix by JS side (when are added the new records, to not reassign an index from existing records) but I do think it will be a better idea to attach identity information.

So basically  there is no identity information supplied within forms (in this case a fieldset) to have an idea for a collection item which records is referring from database. And of course, so at persist Doctrine to take into consideration the existing identity to match collection item with database record.

**Comment by TomHAnderson** [comment]
Closing due to age.
